core/variables.js

```diff
@@ -232,8 +232,9 @@ Blockly.Variables.generateUniqueName = function(workspace) {
  * @param {function(?string=)=} opt_callback A callback. It will
  *     be passed an acceptable new variable name, or null if change is to be
  *     aborted (cancel button), or undefined if an existing variable was chosen.
+ * @param {?string} opt_type Optional type of variable, like 'string' or 'list'.
```

Comment:
As discussed offline, from the user's perspective they might be using BLockly.Variables.CreateVariable and it would benefit them to know about the empty string and what it means. So that has been added to the documentation.